src/theory/uf/equality_engine.cpp

```diff
@@ -1014,16 +1013,21 @@ void EqualityEngine::explainEquality(TNode t1, TNode t2, bool polarity, std::vec
   }
 }
 
-void EqualityEngine::explainPredicate(TNode p, bool polarity, std::vector<TNode>& assertions, EqProof * eqp) const {
-  Debug("equality") << d_name << "::eq::explainPredicate(" << p << ")" << std::endl;
+void EqualityEngine::explainPredicate(TNode p, bool polarity,
+                                      std::vector<TNode>& assertions,
+                                      std::shared_ptr<EqProof> eqp) const {
```

Comment:
It seems like `std::shared_ptr<EqProof> eqp` should be a `std::weak_ptr` since `explainPredicate` is not taking (shared) ownership.